sympy/logic/algorithms/lra_theory.py

```diff
@@ -0,0 +1,667 @@
+from sympy.solvers.solveset import linear_eq_to_matrix
+from sympy.matrices.dense import eye
+from sympy.assumptions.relation.binrel import AppliedBinaryRelation
+from sympy.assumptions.ask import Q
+from sympy.core import Dummy
+from sympy.core.mul import Mul
+from sympy.core.add import Add
+from sympy.core.relational import Eq
+from sympy import SYMPY_DEBUG
+
+
+def sep_const_coeff(expr):
+    if isinstance(expr, Add):
+        return expr, 1
+
+    if isinstance(expr, Mul):
+        coeffs = expr.args
+    else:
+        coeffs = [expr]
+
+    var, const = [], []
+    for c in coeffs:
+        if c.is_constant():
+            const.append(c)
+        else:
+            var.append(c)
+    return Mul(*var), Mul(*const)
+
+
+def list_terms(expr):
+    if not isinstance(expr, Add):
+        return [expr]
+
+    return expr.args
+
+
+def sep_const_terms(expr):
+    if isinstance(expr, Add):
+        terms = expr.args
+    else:
+        terms = [expr]
+
+    var, const = [], []
+    for t in terms:
+        if t.is_constant():
+            const.append(t)
+        else:
+            var.append(t)
+    return sum(var), sum(const)
+
+
+def standardize_binrel(prop):
+    assert prop.function in [Q.ge, Q.gt, Q.le, Q.lt, Q.eq]
+
+    expr = prop.lhs - prop.rhs
+    if prop.function in [Q.ge, Q.gt]:
+        expr = -expr
+    var, const = sep_const_terms(expr)
+
+    if prop.function == Q.eq:
+        return Q.eq(var, const)
+    if prop.function in [Q.gt, Q.lt]:
+        return Q.lt(var, const)
+    else:
+        return Q.le(var, const)
+
+
+class Boundry:
```

Comment:
This could just use a dataclass https://docs.python.org/3/library/dataclasses.html